modules/core/04-channel/keeper/keeper.go

```diff
@@ -571,6 +572,30 @@ func (k Keeper) DeleteUpgradeTimeout(ctx sdk.Context, portID, channelID string)
 	store.Delete(host.ChannelUpgradeTimeoutKey(portID, channelID))
 }
 
+// ValidateProposedUpgradeFields validates the proposed upgrade fields against the existing channel.
+// It returns an error if the following constraints are not met:
+// - there exists at least one valid proposed change to the existing channel fields
+// - the proposed order is a subset of the existing order
+// - the proposed connection hops do not exist
+// - the proposed version is non-empty (checked in ModifiableUpgradeFields.ValidateBasic())
+// - the proposed connection hops are not open
+func (k Keeper) ValidateProposedUpgradeFields(ctx sdk.Context, proposedUpgrade types.ModifiableUpgradeFields, existingChannel types.Channel) error {
+	currentFields := types.ModifiableUpgradeFields{
+		Ordering:       existingChannel.Ordering,
+		ConnectionHops: existingChannel.ConnectionHops,
+		Version:        existingChannel.Version,
+	}
+	if reflect.DeepEqual(proposedUpgrade, currentFields) {
+		return errorsmod.Wrap(types.ErrChannelExists, "existing channel end is identical to proposed upgrade channel end")
+	}
+
+	if !currentFields.Ordering.SubsetOf(proposedUpgrade.Ordering) {
+		return errorsmod.Wrap(types.ErrInvalidChannelOrdering, "channel ordering must be a subset of the new ordering")
+	}
+
+	return k.connectionKeeper.CheckIsOpen(ctx, proposedUpgrade.ConnectionHops[0])
```

Comment:
Personally I think I prefer just doing the check explicitly in here rather than adding another expected keeper interface method but happy to keep this as is if there's consensus on adding this instead. 
My vote would be for doing the check directly here.

Reply:
The idea is that this could be used in many other places, it also covers the not found case. Quickly looking it looks like we have around 7 other duplicate checks.

My preference would be to re-use this helper

Reply:
yeah there was a bit of bike-shedding over the name/scope of this function between us. Initially we had it as `IsOpen` with a single boolean return but that didn't make it as self contained as we'd like. Returning the error allowed it to basically also check for any required error conditions and return them, simplifying the validation, hence the rename to `CheckIsOpen`.

Cian did say this function might not fly with everyone though :smile:

Reply:
I'm happy to just remove the function entirely and do the explicit checking where needed! 👍